Zend/zend_exceptions.c

```diff
@@ -45,7 +45,7 @@ ZEND_API zend_class_entry *zend_ce_unhandled_match_error;
 /* Internal pseudo-exception that is not exposed to userland. */
 static zend_class_entry zend_ce_unwind_exit;
 
-ZEND_API void (*zend_throw_exception_hook)(zval *ex);
+ZEND_API void (*zend_throw_exception_hook)(zend_object *ex);
```

Comment:
Can we actually do this change?

Reply:
Yes, but this should get a note in UPGRADING.INTERNALS.